pulsar/consumer_partition.go

```diff
@@ -109,9 +129,11 @@ func newPartitionConsumer(parent Consumer, client *client, options *partitionCon
 		eventsCh:       make(chan interface{}, 3),
 		queueSize:      int32(options.receiverQueueSize),
 		queueCh:        make(chan []*message, options.receiverQueueSize),
+		startMessageID: options.startMessageID,
 		connectedCh:    make(chan struct{}),
 		messageCh:      messageCh,
 		closeCh:        make(chan struct{}),
+		clearQueueCh:   make(chan func(id *messageID)),
```

Comment:
Are we ok with this channel being unbuffered?

Reply:
I think it's ok since the dispatcher routine is not blocked on writing on the application channel